src/dnsmasq_interface.c

```diff
@@ -2570,95 +2570,86 @@ void print_flags(const unsigned int flags)
 	free(flagstr);
 }
 
-static const char *reply_status_str[QUERY_REPLY_MAX+1] = {
-	"UNKNOWN",
-	"NODATA",
-	"NXDOMAIN",
-	"CNAME",
-	"IP",
-	"DOMAIN",
-	"RRNAME",
-	"SERVFAIL",
-	"REFUSED",
-	"NOTIMP",
-	"OTHER",
-	"DNSSEC",
-	"NONE",
-	"BLOB",
-	"MAX"
-};
-
 static void _query_set_reply(const unsigned int flags, const enum reply_type reply,
                              const union all_addr *addr,
                              queriesData *query, const struct timeval response,
                              const char *file, const int line)
 {
+	enum reply_type new_reply = REPLY_UNKNOWN;
 	// If reply is set, we use it directly instead of interpreting the flags
 	if(reply != 0)
 	{
-		query->reply = reply;
+		new_reply = reply;
 	}
 	// else: Iterate through possible values by analyzing both the flags and the addr bits
 	else if(flags & F_NEG ||
-	   (flags & F_NOERR && !(flags & (F_IPV4 | F_IPV6))) || // <-- FTL_make_answer() when no A or AAAA is added
-	   force_next_DNS_reply == REPLY_NXDOMAIN ||
-	   force_next_DNS_reply == REPLY_NODATA)
+	        (flags & F_NOERR && !(flags & (F_IPV4 | F_IPV6))) || // <-- FTL_make_answer() when no A or AAAA is added
+	        force_next_DNS_reply == REPLY_NXDOMAIN ||
+	        force_next_DNS_reply == REPLY_NODATA)
 	{
 		if(flags & F_NXDOMAIN || force_next_DNS_reply == REPLY_NXDOMAIN)
 			// NXDOMAIN
-			query->reply = REPLY_NXDOMAIN;
+			new_reply = REPLY_NXDOMAIN;
 		else
 			// NODATA(-IPv6)
-			query->reply = REPLY_NODATA;
+			new_reply = REPLY_NODATA;
 	}
 	else if(flags & F_CNAME)
 		// <CNAME>
-		query->reply = REPLY_CNAME;
+		new_reply = REPLY_CNAME;
 	else if(flags & F_REVERSE)
 		// reserve lookup
-		query->reply = REPLY_DOMAIN;
+		new_reply = REPLY_DOMAIN;
 	else if(flags & F_RRNAME)
 		// TXT query
-		query->reply = REPLY_RRNAME;
+		new_reply = REPLY_RRNAME;
 	else if((flags & F_RCODE && addr != NULL) || force_next_DNS_reply == REPLY_REFUSED)
 	{
 		if((addr != NULL && addr->log.rcode == REFUSED)
 		   || force_next_DNS_reply == REPLY_REFUSED )
 		{
 			// REFUSED query
-			query->reply = REPLY_REFUSED;
+			new_reply = REPLY_REFUSED;
 		}
 		else if(addr != NULL && addr->log.rcode == SERVFAIL)
 		{
 			// SERVFAIL query
-			query->reply = REPLY_SERVFAIL;
+			new_reply = REPLY_SERVFAIL;
 		}
```

Comment:
To implement `reply_NOTIMP`, we'll add here:
```suggestion
		}
		else if(addr != NULL && addr->log.rcode == NOTIMP)
		{
			// NOT IMPLEMENTED reply
			query->reply = REPLY_NOTIMP;
			new_reply = REPLY_NOTIMP;
		}
```

Reply:
Added by https://github.com/pi-hole/FTL/pull/1325